Finalize 1.21.5 FRAPI port and other improvements

PepperCode1 opened the pull request:
- Allow retrieving `SpriteFinder`s from `ErrorCollectingSpriteGetter`, `SpriteLoader.StitchResult`, and `SpriteAtlasManager.AtlasPreparation`
- Add injected interfaces to allow retrieving `SpriteFinder`s directly
- Ensure lazily initialization of sprite finders is completely thread safe
- Provide a default implementation for `MutableQuadView/QuadEmitter.spriteBake`
- Add `ModelBakeSettingsHelper`, which has methods for creating `ModelBakeSettings` from an arbitrary matrix, multiplying two `ModelBakeSettings`, and converting a `ModelBakeSettings` into a `QuadTransform`
- Make fast smooth lighting computation for cubic quads aligned to their light face work correctly regardless of vertex order
- Remove `ApitStatus.Experimental` annotations
- Deprecate `Renderer.emitBlockModelPartQuads` for removal
- Fix `GeometryBakedModel.emitQuads` not applying instance AO boolean when using `MeshBakedGeometry`
- Add `RenderLayerHelper.movingDelegate/entityDelegate` helpers and use them where appropriate

PepperCode1 committed:
Allow using sprite finder during model baking

- Allow retrieving sprite finders from ErrorCollectingSpriteGetter, SpriteLoader.StitchResult, and SpriteAtlasManager.AtlasPreparation via new interface injections
- Deprecate SpriteFinder.get in favor of interface injected method for consistency with other methods
- Ensure lazily initialization of sprite finders is completely thread safe
- Provide a default implementation for MutableQuadView/QuadEmitter.spriteBake
- Improve some documentation

modmuss50 commented:
Not sure if it was ready to be reviewed or not as its still marked as draft. When you are ready this can be moved to last call.

PepperCode1 committed:
Add ModelBakeSettingsHelper

- Respect passed ModelBakeSettings in testmod custom Geometry implementations
- Add axis block state property to testmod octagonal column block and rotate model to test ModelBakeSettings application

PepperCode1 commented:
Feel free to review, but there are at least 3 separate changes left to make on top of the 2 currently in the PR.

modmuss50 commented:
Ah no worries at all, let me know when you are ready for me to have another look. Thanks :)

PepperCode1 committed:
Make AoCalculator.vanillaFullFace work regardless of vertex order

- Use vertex map for cubic faces during enhanced light as well now that it works regardless of vertex order and is faster than using weights
- Fix typos in AoFaceData

PepperCode1 committed:
Finalization

- Remove Experimental annotations
- Deprecate Renderer.emitBlockModelPartQuads for removal
- Fix GeometryBakedModel.emitQuads not applying instance AO boolean
- Add RenderLayerHelper.movingDelegate/entityDelegate and use them where appropriate
- Clarify some javadoc

modmuss50 commented on fabric-renderer-api-v1/src/client/java/net/fabricmc/fabric/mixin/renderer/client/sprite/ErrorCollectingSpriteGetterMixin.java:
```diff
@@ -0,0 +1,27 @@
+import net.fabricmc.fabric.api.renderer.v1.sprite.FabricErrorCollectingSpriteGetter;
+
+@Mixin(ErrorCollectingSpriteGetter.class)
+interface ErrorCollectingSpriteGetterMixin extends FabricErrorCollectingSpriteGetter {
```
Where is FabricErrorCollectingSpriteGetter.spriteFinder implimented? Or is it supposed to always throw the `UnsupportedOperationException` when used via ErrorCollectingSpriteGetter?

modmuss50 replied:
Seems this is the only implimentation of this interface, and no where (other than javadocs) refrences it.

PepperCode1 replied:
It is implemented in `BakedModelManager1Mixin`, as vanilla's only implementation of the interface is an inline class in `BakedModelManager`.

modmuss50 replied:
Ah yeah, intelij is being silly and not telling me this. Sorry!
